### PR TITLE
[Android] Remove AndroidExtra header

### PR DESCRIPTION
**Pull request by joseluismarti**
## Description
This header was used to incorporate defines that were missing in early NDKs < r9b

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/android/activity/AndroidKey.cpp [hunk 1]
```diff
@@ -8,15 +8,12 @@
 
 #include "AndroidKey.h"
 
-#include "AndroidExtra.h"
 #include "ServiceBroker.h"
 #include "XBMCApp.h"
-#include "input/Key.h"
```
Comment (fuzzard) [inline xbmc/platform/android/activity/AndroidKey.cpp:14]: I couldn't track down where the transitive include for https://github.com/xbmc/xbmc/blob/a03dede299ed07de5a8e6e9462fff62858acbf87/xbmc/input/XBMC_keysym.h is coming from, but I think it should be added in explicitly for the use of XBMCK_* defines

**Review by fuzzard — COMMENTED** [review]
I dont like leaving things to transitive includes. We've had issues in the past where something gets changed and breaks happen all over the place because of transitive inclusion, and if we ever go further with pch usage, its a pain to unravel issues.

My 2 comments look to be the only things i would change, all the other changes you make seem fine.

File: xbmc/platform/android/activity/AndroidKey.h [hunk 2]
```diff
@@ -8,10 +8,6 @@
 
 #pragma once
 
-#include <stdint.h>
```
[the remark this answers is not in the document]
Comment (joseluismarti) [reply xbmc/platform/android/activity/AndroidKey.h:11]: Although `stdint.h` is included in `android/input.h`, you are right that it is preferable to include it explicitly.

File: xbmc/platform/android/activity/AndroidKey.cpp (hunk 1 above)
Comment (joseluismarti) [reply xbmc/platform/android/activity/AndroidKey.cpp:14]: You are right, we should not rely on headers being included transitively. We must include a header if it's explicitly needed in that file.

**Pull request merged**